ExcelReaderAPI/Services/UserAuthService.cs

```diff
@@ -0,0 +1,211 @@
+﻿using System.Data.SqlClient;
+using System.IdentityModel.Tokens.Jwt;
+using System.Security.Claims;
+using System.Security.Cryptography;
+using System.Text;
+using ExcelReaderAPI.Models;
+using Microsoft.IdentityModel.Tokens;
+
+namespace ExcelReaderAPI.Services
+{
+    public class UserAuthService
+    {
+
+        private readonly DatabaseHelper _databaseHelper;
+        private readonly IConfiguration _configuration;
+
+
+        public UserAuthService(IConfiguration configuration, DatabaseHelper databaseHelper)
+        {
+            _databaseHelper = databaseHelper;
+            _configuration = configuration;
+
+        }
+
+        private void CreatePasswordHash(string password, out byte[] passwordHash, out byte[] passwordSalt)
+        {
+            using (var hmac = new HMACSHA512())
+            {
+                passwordSalt = hmac.Key;
+                passwordHash = hmac.ComputeHash(System.Text.Encoding.UTF8.GetBytes(password));
+            }
+        }
+
+        private bool VerifyPasswordHash(string password, byte[] passwordHash, byte[] passwordSalt)
+        {
+            using (var hmac = new HMACSHA512(passwordSalt))
```

Comment:
these kind of functions should be somewhere in helper class

Reply:
done, Alhamdulillah